arches/app/tasks.py

```diff
@@ -629,6 +629,108 @@ def reverse_etl_load(loadid):
     module.reverse_load(loadid)
 
 
+@shared_task
+def update_resource_instance_data_based_on_graph_diff(
+    initial_graph, updated_graph, user_id
+):
+    logger = logging.getLogger(__name__)
+    user = User.objects.get(id=user_id)
+
+    try:
+        # delete extra tiles on nodegroup cardinality change
+        initial_nodegroup_ids_to_cardinality = {}
+        for initial_nodegroup in initial_graph["nodegroups"]:
+            initial_nodegroup_ids_to_cardinality[initial_nodegroup["nodegroupid"]] = (
+                initial_nodegroup["cardinality"]
+            )
+
+        nodegroups_with_extra_tiles = []
+        for updated_nodegroup in updated_graph["nodegroups"]:
+            if (
+                updated_nodegroup["cardinality"] == "1"
+                and initial_nodegroup_ids_to_cardinality.get(
+                    updated_nodegroup["nodegroupid"]
+                )
+                == "n"
+            ):
+                nodegroups_with_extra_tiles.append(updated_nodegroup["nodegroupid"])
+
+        for tile in models.TileModel.objects.exclude(sortorder=0).filter(
+            nodegroup_id__in=nodegroups_with_extra_tiles
+        ):
+            tile.delete()
```

Comment:
I think that we may also need to delete tiles where nodegroupids may have changed; for example if a node is changed from being a "collector" to not and then one of it's parents is made a "collector" then we will need to delete all tiles for the original nodegroup.  I think that we can simply delete tiles for nodegroupids that don't match current nodegroups.  What do you think?

This will cascade to child tiles, correct?

Reply:
another Q here: what about when a new nodegroup is introduced into an existing tree of nodes (ie flip an intervening node in a tree to "is collector")? this then orphans tiles in a sense as it can result in tiles whose specified parent is not of the right parent nodegroup.  The simple solution to just delete these tiles as well (and their children) I guess